### ISSUE TITLE
Memory validation

### ISSUE THREAD

**Issue by cyberblast** [ScreepsOCS/screeps.behaviour-action-pattern#595]
We need a reliable system of memory validation. 
Trusting in one time clean up events doesn't seem to be sufficient. 

First I thought about a "global" memory parsing solution which would parse the whole memory at once, but that would maybe be quite a mess. 

However. 
Most issues are with our new tasks. 
We need to validate those entries. Maybe not every tick. 
Most tasks do already have a validation implemented, triggered by an event. But we should do that occasionally as well.

**Comment by karlthepagan** [comment]
I favor segmenting queues in order to minimize user errors when manipulating spawn queues.

User api should have well defined places to be safely edited.

**Comment by cyberblast** [comment]
Hmm, you mean like a wrapper class for spawnQueues? 
Removal of an entry could have a look if it has a "task" and use a callback to clean up task memory.

**Comment by karlthepagan** [comment]
Yes that could work. Perhaps using the task cache to connect the spawn queue entries to an object that will callback when the entry is changed unexpectedly?

Then we only have to validate a limited number of times.
